packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -71,104 +96,236 @@ export function matches(element: Element, selector: AST): boolean {
       }
 
       return text.length >= minLength;
+    } else if (selector.name === 'matches-path') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      // Get the window object from the element's ownerDocument
+      const window = element.ownerDocument?.defaultView;
+      if (!window) {
+        return false;
+      }
+
+      // Get both pathname and search (query parameters)
+      const path = window.location.pathname;
+      const search = window.location.search;
+      const fullUrl = path + search;
+      const regex = parseRegex(argument);
+      return regex.test(fullUrl);
+    } else if (selector.name === 'matches-attr') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const indexOfEqual = argument.indexOf('=');
+      let namePattern: string;
+      let valuePattern: string | undefined;
+      if (indexOfEqual === -1) {
+        namePattern = argument;
+      } else {
+        namePattern = argument.slice(0, indexOfEqual);
+        valuePattern = argument.slice(indexOfEqual + 1);
+      }
+
+      namePattern = stripsWrappingQuotes(namePattern);
```

Comment:
If we handle quoted keys, do we have to handle arbitrary strings like that?

"x=y"=z